### PR TITLE
docs: update the readme

### LINKED ISSUE #23657: remove unnecessary text

### PR DESCRIPTION
**Pull request by a0m0rajab**
# PR Description

This PR is using a stacked PR logic to make it easier to read and review, to start reviewing it we need to merge them in this order: 
- https://github.com/unifyai/ivy/pull/23654
- https://github.com/unifyai/ivy/pull/23653
- https://github.com/unifyai/ivy/pull/23652

## Related Issue

Close #23657

## Checklist

- [ ] Did you add a function?
- [ ] Did you add the tests?
- [ ] Did you run your tests and are your tests passing?
- [ ] Did pre-commit not fail on any check?
- [ ] Did you follow the steps we provided?

### CODE REVIEW

**Comment by guillesanbri** [comment]
same here, not exactly sure of what's the purpose of the guides.md file :smiley:

**Comment by a0m0rajab** [comment]
I will remove the guides file in the last PR though.

File: README.md
Comment (guillesanbri) [inline README.md:100]: I still think this is too vague, doesn't really explain what the autotuner is, seems to focus around compression when this is just a small step that is only applied in a subset of the transformations we do

File: README.md
Comment (guillesanbri) [inline README.md:109]: I'm inclined to say that this feels a bit off after the characteristics, seems more like an ending for the readme

File: README.md
Comment (guillesanbri) [inline README.md:116]: no caps after comma

File: README.md
Comment (guillesanbri) [inline README.md:115]: Either "You can start by reading our tutorials and demos" or "You can start by reading our tutorials and exploring our demos". I think the first one is better

File: README.md
Comment (guillesanbri) [inline README.md:117]: How to convert you code between frameworks

File: README.md
Comment (guillesanbri) [inline README.md:118]: framework-agnostic, not framework-free, also this links to the unify notebook which is not really about writing code, but rather converting to ivy code automatically

File: README.md
Comment (guillesanbri) [inline README.md:119]: too generic imo

File: README.md
Comment (guillesanbri) [inline README.md:128]: "You can find our documentation in the [Ivy Docs page], which includes"

File: README.md
Comment (guillesanbri) [inline README.md:131]: no space between the title and the description

File: README.md
Comment (guillesanbri) [inline README.md:140]: Would be better to have motivation -> related work -> design -> deep dive (from more general to more in-depth)

Comment (guillesanbri) [inline README.md:140]: I would remove the clear

File: README.md
Comment (guillesanbri) [inline README.md:135]: two extra commas

File: README.md
Comment (guillesanbri) [inline README.md:184]: same with this, I think it's useful to present the value the user can get from transpile (altough this can change and be reduced once we have the full suite of functions)

File: README.md
Comment (guillesanbri) [inline README.md:339]: We should definitely keep the instructions to install ivy in the readme, if someone goes to the repo and just want to install ivy they shouldn't have to look for the instructions in the docs, this should almost be a copy of https://unify.ai/docs/ivy/overview/get_started.html imo

Comment (guillesanbri) [reply README.md:339]: we can reference some sections but the basic installation instructions should definitely be here

**Review by guillesanbri — CHANGES_REQUESTED** [review]
just left a few comments :+1:

File: README.md
Comment (a0m0rajab) [reply README.md:100]: Fixed.

File: README.md
Comment (a0m0rajab) [reply README.md:104]: Fixed.

File: README.md
Comment (a0m0rajab) [reply README.md:107]: done

File: README.md
Comment (a0m0rajab) [reply README.md:115]: done.

File: README.md
Comment (guillesanbri) [inline README.md:110]: same as above with the ":"

File: README.md
Comment (guillesanbri) [inline README.md:112]: typo in "basicss"

File: README.md
Comment (guillesanbri) [inline README.md:114]: caps in here

File: README.md
Comment (guillesanbri) [inline README.md:165]: caps in subtitle, would be good to go through the text after doing changes to ensure spelling and style is consistent :smile:

File: README.md
Comment (guillesanbri) [inline README.md:100]: if the intro says `enables you to` the verbs should reflect that, i.e. `Autotune` instead of `autotunes` and so on

File: README.md
Comment (a0m0rajab) [reply README.md:122]: what do you mean?

**Comment by vedpatwardhan** [comment]
Sorry for jumping in, but I'd suggest creating a branch on a fork instead of the main repo if it's not used for something that requires external contributors (from outside the organization). No worries for this PR, please delete this branch from the main repo once you've merged. Thanks @a0m0rajab 😄

File: README.md
Comment (guillesanbri) [reply README.md:122]: I mean to replace this section with just the installation instructions instead :)

File: README.md
Comment (guillesanbri) [inline README.md:255]: There is no `:` and no space between `Deep dive` and `explains`, also if we capitalize both words in `Related Work` we should do the same in `Deep Dive`.

File: README.md
Comment (guillesanbri) [inline README.md:252]: No `:` after related work

File: README.md
Comment (guillesanbri) [inline README.md:247]: If we are going with bullet points I'd capitalize them and finish all of them with `.`

File: README.md
Comment (guillesanbri) [inline README.md:246]: `Motivation` instead of `Motiviation`

File: README.md
Comment (guillesanbri) [inline README.md:245]: On a more general level, I think the phrasing is a bit inconsistent here for each one of the bullet point, e.g. using `this contextualizes` or `which paints` and then simply enumerating stuff in design, if that makes sense?

File: README.md
Comment (guillesanbri) [inline README.md:1350]: I think we should have a very brief introduction before the collapsed sections here

File: README.md
Comment (guillesanbri) [reply README.md:122]: Furthermore, the installation instructions can maybe be the first thing in the `Getting Started` section above

File: README.md
Comment (guillesanbri) [inline README.md:110]: ":" after the title

File: README.md
Comment (guillesanbri) [inline README.md:103]: Also I think this can be more informative while keeping it short, e.g.

---
- 🔥 **Autotune your model**: Automatically find the optimal framework, compiler infrastructure and hardware for your specific use case using `ivy.autotune`.
- 💥 **Boost model speed**:  Speed up your models by up to 3x using the most efficient ML infrastructure with `ivy.compile`.
- 🔄 **Convert code into any framework**: Use and build on top of any model, library, or device by converting any code from one framework to another using `ivy.transpile`.
- ⚒️ **Write framework-agnostic code**: Write your code once in ivy and then choose the most appropriate ML framework as the backend to leverage all the benefits and tools.
---

File: README.md
Comment (a0m0rajab) [reply README.md:122]: As for the getting started, we can add one or two links later.

File: README.md
Comment (a0m0rajab) [reply README.md]: I think we still need to have something like that, or just move the logos under the text and name it as supported libraries.

File: README.md
Comment (a0m0rajab) [reply README.md:106]: changed it to: connect

File: README.md
Comment (a0m0rajab) [reply README.md:245]: Done

Comment (a0m0rajab) [reply README.md:245]: Can you elaborate?

File: README.md
Comment (a0m0rajab) [reply README.md:1346]: would you like to have this on a new PR? just to minimise the scope here and move faster?

File: README.md
Comment (a0m0rajab) [reply README.md]: I feel this should be like a hook rather than too informative, even if we are talking to engineer, getting them curious with short time was the main, I changed it to your suggestions though.

File: README.md
Comment (guillesanbri) [reply README.md:121]: I think in a lot of cases it will be the other way around, the readme is not only for new people who come across the project, but also for people who know about the project and want clear information. Installation instructions are usually first class citizens in the readme I'd say

- https://github.com/pytorch/pytorch#installation
- https://github.com/tensorflow/tensorflow#install (almost the only thing they have)
- https://github.com/google/jax#installation

A lot of people will install it and then go to the examples or the docs directly imo

File: README.md
Comment (guillesanbri) [reply README.md:113]: Still unchanged :sweat_smile:

File: README.md
Comment (guillesanbri) [reply README.md:1345]: I feel the collpased sections are too isolated at the moment. Specially being after all the other collapsed sections, I think we should have the introductions before merging these changes to give this section more structure

File: README.md
Comment (guillesanbri) [reply README.md]: Thanks! Yeah I get your point, I don't really think the current ones are too informative tho

Comment (a0m0rajab) [reply README.md]: Sure, thanks for that! I will count this as resolved.

File: README.md
Comment (a0m0rajab) [reply README.md:1345]: Thanks for that! added one line explainer.

File: README.md
Comment (a0m0rajab) [reply README.md]: Thanks, I added that!

File: README.md
Comment (a0m0rajab) [reply README.md]: Sure, I removed it, yet I still feel it's needed.

File: README.md
Comment (a0m0rajab) [reply README.md]: Done!

**Review by guillesanbri — COMMENTED** [review]
Just made some minor changes myself, other than this (and we should be good to merge prob):
- [ ] Can you lower the ivy logo a bit in the "Using Ivy" subtitle? it's a bit too far from the centre of the line imo
- [ ] Not sure if you are doing this already but just in case, we need to build the docs locally to ensure that the rendered version of the readme that shows up in the doc is working fine :+1:

**Comment by a0m0rajab** [comment]
@guillesanbri thanks for the comment as for the logo I could not play with that, since GitHub does not allow CSS or Style in the markdown, for security reasons. 

As for the second point, I did not use any RST specific tags which should not break the files, If you want I might ask @KareemMAX to show me the docs workflow and build it.

**Comment by KareemMAX** [comment]
> @guillesanbri thanks for the comment as for the logo I could not play with that, since GitHub does not allow CSS or Style in the markdown, for security reasons.
> 
> As for the second point, I did not use any RST specific tags which should not break the files, If you want I might ask @KareemMAX to show me the docs workflow and build it.

Sphinx uses an extended version of markdown (MyST). So ideally if Github renders everything fine so should Sphinx. But I'll contact @a0m0rajab later today to build the docs

**Comment by guillesanbri** [comment]
> > @guillesanbri thanks for the comment as for the logo I could not play with that, since GitHub does not allow CSS or Style in the markdown, for security reasons.
> > As for the second point, I did not use any RST specific tags which should not break the files, If you want I might ask @KareemMAX to show me the docs workflow and build it.
> 
> Sphinx uses an extended version of markdown (MyST). So ideally if Github renders everything fine so should Sphinx. But I'll contact @a0m0rajab later today to build the docs

Yeah I imagine it should work fine but worth doing a sanity check just in case :+1:

**Comment by guillesanbri** [comment]
> @guillesanbri thanks for the comment as for the logo I could not play with that, since GitHub does not allow CSS or Style in the markdown, for security reasons.

I think the way we did it was by changing the white space in the image, a bit hacky but it did the trick :sweat_smile:

**Comment by guillesanbri** [comment]
Looks good to go, only thing is that the refactor from `compile` to `trace_graph` has been merged, so there are conflicts + the readme has to be updated as well :sweat_smile:

**Comment by a0m0rajab** [comment]
@guillesanbri thanks for that, merge conflict resolved and we tested it with the local doc builder, we had an issue there which took some time to debug but it's working right now.

**Review by guillesanbri — APPROVED** [review]

**Pull request merged**
